File: aws_fargate/fetchTeeTimes-CHRONO-fargate/TeeTime.ts
export class TeeTime {
  start_datetime: string;
  players_available: number;
  available_participants: number[];
  holes: number;
  price: number;
  booking_link: string;
  booking_links: { [key: number]: string };
  tee_time_id: string;

  constructor(
    start_datetime: string,
    players_available: number,
    available_participants: number[],
    holes: number,
    price: number,
    booking_link: string,
    booking_links: { [key: number]: string },
    tee_time_id: string
  ) {
    this.start_datetime = start_datetime;
    this.players_available = players_available;
    this.available_participants = available_participants;
    this.holes = holes;
    this.price = price;
    this.booking_link = booking_link;
    this.booking_links = booking_links;
    this.tee_time_id = tee_time_id;
  }

  // Helper method to create a TeeTime from a plain object
  static fromObject(obj: any): TeeTime {
    return new TeeTime(
      obj.start_datetime,
      obj.players_available,
      obj.available_participants,
      obj.holes,
      obj.price,
      obj.booking_link,
      obj.booking_links,
      obj.tee_time_id
    );
  }

  // Helper method to convert TeeTime to a plain object
  toObject(): any {
    return {
      start_datetime: this.start_datetime,
      players_available: this.players_available,
      available_participants: this.available_participants,
      holes: this.holes,
      price: this.price,
      booking_link: this.booking_link,
      booking_links: this.booking_links,
      tee_time_id: this.tee_time_id
    };
  }
} 

File: aws_fargate/chrono/tee_time.py
"""TeeTime data class for ChronoGolf integration"""
from typing import Dict, Any, List
from dataclasses import dataclass, field


@dataclass
class TeeTime:
    """Represents a single tee time slot with all booking details"""
    
    start_datetime: str
    players_available: int
    available_participants: List[int]
    holes: int
    price: float
    booking_link: str
    booking_links: Dict[int, str]
    tee_time_id: str

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> 'TeeTime':
        """Create a TeeTime instance from a dictionary
        
        Args:
            data: Dictionary containing tee time data
            
        Returns:
            TeeTime instance
        """
        return cls(
            start_datetime=data['start_datetime'],
            players_available=data['players_available'],
            available_participants=data['available_participants'],
            holes=data['holes'],
            price=data['price'],
            booking_link=data['booking_link'],
            booking_links=data['booking_links'],
            tee_time_id=data['tee_time_id']
        )

    def to_dict(self) -> Dict[str, Any]:
        """Convert TeeTime to a dictionary
        
        Returns:
            Dictionary representation of the TeeTime
        """
        return {
            'start_datetime': self.start_datetime,
            'players_available': self.players_available,
            'available_participants': self.available_participants,
            'holes': self.holes,
            'price': self.price,
            'booking_link': self.booking_link,
            'booking_links': self.booking_links,
            'tee_time_id': self.tee_time_id
        }

